### PR TITLE
Apim 7165 portal publish kafka api

### PR DESCRIPTION
**Pull request by illichso**
## Issue

https://gravitee.atlassian.net/browse/APIM-7165

## Description

As an API publisher

I want to publish my Kafka API to the developer portal

so that subscribers can view documentation and subscribe to my API.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-zoukwfphis.chromatic.com)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java [hunk 1]
```diff
@@ -261,6 +285,9 @@ public String getApiEntrypointsListenerType(GenericApiEntity genericApiEntity) {
         ) {
             return "HTTP";
         }
+        if(genericApiEntity.getDefinitionVersion() == DefinitionVersion.V4 && genericApiEntity instanceof NativeApiEntity){
```
Comment (ThibaudAV) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java]: I think we also need to check type like if NativeApiEntity.type is Kafka

Comment (ThibaudAV) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java]: oups still draft sry i saw a notification. i will check it later ;)

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java [hunk 2]
```diff
@@ -261,6 +277,9 @@ public String getApiEntrypointsListenerType(GenericApiEntity genericApiEntity) {
         ) {
             return "HTTP";
         }
+        if (genericApiEntity.getDefinitionVersion() == DefinitionVersion.V4 && genericApiEntity instanceof NativeApiEntity) {
+            return "KAFKA";
+        }
```
Comment (ThibaudAV) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java]: This is necessary ? For me the rule here it's the same for any V4 api we check the first listener type.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/v4/nativeapi/NativeApiModel.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    private Set<String> groups;
+    private Visibility visibility;
+    private Lifecycle.State state;
+    private PrimaryOwnerEntity primaryOwner;
+    private String picture;
+    private Set<String> categories;
+    private Map<String, String> metadata = new HashMap<>();
+    private ApiLifecycleState lifecycleState;
+    private boolean disableMembershipNotifications;
+    private Failover failover;
```
Comment (ThibaudAV) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/v4/nativeapi/NativeApiModel.java]:
```suggestion
    private String id;
    private String name;
    private String apiVersion;
    private DefinitionVersion definitionVersion;
    private ApiType type;
    private Date deployedAt;
    private Date createdAt;
    private Date updatedAt;
    private String description;
    private Set<String> tags = new HashSet<>();
    private List<NativeListener> listeners;
    private List<NativeEndpointGroup> endpointGroups;
    private List<Property> properties = new ArrayList<>();
    private List<Resource> resources = new ArrayList<>();
    private NativeApiServices services;
    private Set<String> groups;
    private Visibility visibility;
    private Lifecycle.State state;
    private PrimaryOwnerEntity primaryOwner;
    private String picture;
    private Set<String> categories;
    private Map<String, String> metadata = new HashMap<>();
    private ApiLifecycleState lifecycleState;
    private boolean disableMembershipNotifications;
```

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java [hunk 4]
```diff
@@ -155,6 +157,15 @@ private List<ApiEntrypointEntity> getEntrypoints(
                         .stream()
                 )
                 .toList();
+        } else if (genericApiEntity.getDefinitionVersion() == DefinitionVersion.V4 && genericApiEntity instanceof NativeApiEntity api) {
+            return api
+                .getListeners()
+                .stream()
+                .map(listener -> {
+                    var kafkaListener = (KafkaListener) listener;
+                    return getKafkaNativeApiEntrypointEntity(kafkaListener.getHost(), kafkaListener.getPort(), tagEntrypoints);
```
Comment (ThibaudAV) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java]:
```suggestion
// FIXME: use the kafkaPort config from properties when available (like tcpPort)
                    return getKafkaNativeApiEntrypointEntity(kafkaListener.getHost(), 9092), tagEntrypoints);
```

the getPort here must not be used. it's for another concept, but not necessary for 4.6.

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java [hunk 5]
```diff
@@ -155,6 +157,15 @@ private List<ApiEntrypointEntity> getEntrypoints(
                         .stream()
                 )
                 .toList();
+        } else if (genericApiEntity.getDefinitionVersion() == DefinitionVersion.V4 && genericApiEntity instanceof NativeApiEntity api) {
+            return api
+                .getListeners()
+                .stream()
+                .map(listener -> {
+                    var kafkaListener = (KafkaListener) listener;
```
Comment (ThibaudAV) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java:166]: what about add a `if (listener instanceof KafkaListener httpListener) {` ? in the future we ca also have MQTTListener

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java (hunk 2 above)
Comment (illichso) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java]: Updated

File: gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/v4/nativeapi/NativeApiModel.java (hunk 3 above)
Comment (illichso) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-model/src/main/java/io/gravitee/rest/api/model/v4/nativeapi/NativeApiModel.java]: Updated

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java (hunk 5 above)
Comment (illichso) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java:166]: Done

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java (hunk 1 above)
Comment (illichso) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java]: Added filters

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java (hunk 4 above)
Comment (illichso) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiEntrypointServiceImpl.java]: Done

**Review by ThibaudAV — APPROVED** [review]

**Pull request merged**